src/server/ua_server_worker.c

```diff
@@ -563,6 +569,50 @@ UA_StatusCode UA_Server_run_startup(UA_Server *server) {
         result |= nl->start(nl, server->config.logger);
     }
 
+
+#ifdef UA_ENABLE_DISCOVERY_MULTICAST
+    if (server->config.applicationDescription.applicationType == UA_APPLICATIONTYPE_DISCOVERYSERVER) {
+
+        char *appName = (char *)malloc(server->config.mdnsServerName.length +1);
+        memcpy(appName, server->config.mdnsServerName.data, server->config.mdnsServerName.length);
+        appName[server->config.mdnsServerName.length] = '\0';
+
+        for(size_t i = 0; i < server->config.networkLayersSize; i++) {
```

Comment:
Split the inner part of the for-loop in a separate method.
And add comments. What does the code do? Why for every networklayer? Why here?